### PR TITLE
did correct wrong dates in 2000, corrected url for plots/species

### PR DESCRIPTION
**Pull request by katrintirok**
Hi Ethan,

I noticed that the portal teaching data has a 31st April 2000 and 31st September 2000. When trying to combine year, month, day to a date in python using pd.to_datetime(...) it throughs an error.

I added code to replace those dates with 1st of May and 1st of October 2000, resp., to the create_portal_teaching_dataset.py script and replaced all cvs, json and sqlite files accordingly.

Further, the urls to the species.txt and plots.txt files in the python script did not work, I updated these urls.

How does the data set get updated on figshare?, because I use the figshare link to download data when teaching the Data Carpentry ecology lesson.

Kind Regards
Katrin

### CODE REVIEW

File: create_portal_teaching_dataset.py [hunk 1]
```diff
@@ -1,3 +1,10 @@
+# -*- coding: utf-8 -*-
+"""
+Updated on Mon Mar  5 21:56:52 2018 by Katrin Tirok
+Corrected one wrong date in surveys table
+Corrected url for download of species and plot tables
+"""
+
```
Comment (ethanwhite) [inline create_portal_teaching_dataset.py]: This information belongs in the commit message not the file. Unless there's a reason to specific to provide the encoding it shouldn't be included and typically `utf-8` is the default. In short, I think we can just remove this seven line change.

File: create_portal_teaching_dataset.py [hunk 2]
```diff
@@ -13,9 +20,23 @@
...
+surveys.loc[(surveys.year==2000) & (surveys.month==4) & (surveys.day==31),'day'] = 1
+surveys.loc[(surveys.year==2000) & (surveys.month==4) & (surveys.day==1),'month'] = 5
+surveys.loc[(surveys.year==2000) & (surveys.month==9) & (surveys.day==31), 'day'] = 1
+surveys.loc[(surveys.year==2000) & (surveys.month==9) & (surveys.day==1), 'month'] = 10
 
 # Clean Species Table
-species = pd.read_csv("http://wiki.ecologicaldata.org/sites/default/files/portal_species.txt",
+# old url:
+# url = "http://wiki.ecologicaldata.org/sites/default/files/portal_species.txt"
+# current url (for species and plot table):
```
Comment (ethanwhite) [inline create_portal_teaching_dataset.py]: I don't think we need the three commented lines here. Let's just fix the url and note the fix in the commit message.

File: create_portal_teaching_dataset.py [hunk 3]
```diff
@@ -29,7 +50,9 @@
 species = pd.concat([species_id, split_names, taxa], axis=1)
 
 # Clean Plots Table
-plots = pd.read_csv("http://wiki.ecologicaldata.org/sites/default/files/portal_plots.txt",
+# old url:
+# url = "http://wiki.ecologicaldata.org/sites/default/files/portal_plots.txt"
```
Comment (ethanwhite) [inline create_portal_teaching_dataset.py]: Likewise, drop the comments.

File: create_portal_teaching_dataset.py [hunk 4]
```diff
@@ -13,9 +20,23 @@
                inplace=True)
 surveys.replace({'species_id': {'NA': 'NL'}, 'sex': {'P': float('nan'), 'R': float('nan'), 'Z': float('nan')}}, inplace=True)
 
+# correct wrong date, pd.to_datetime throughs an error when combining year, month, day to a date
+# in the year 2000 there is a 31st April and a 31st of September, 
+# perhaps these should be 1st of May and 1st of October, resp.
+# to check:
+# pd.to_datetime(surveys[['year', 'month', 'day']])
+# replace wrong dates:
```
Comment (ethanwhite) [inline create_portal_teaching_dataset.py]: This comment block can be reduced to:
`# Replace invalid dates`

**Review by ethanwhite — CHANGES_REQUESTED** [review]
Thanks @katrintirok! This all looks great. There are a few changes in the `create_portal_teaming_dataset.py` that I think should be removed and I've noted them in-line.

The upload to figshare is still manual. Once you update the `.py` I'll get this merged and update the files on figshare.

**Comment by katrintirok** [comment]
@ethanwhite thanks for the review, I changed the python script accordingly. Please check again and let me know when anything else needs to be changed.

**Comment by ethanwhite** [comment]
Thanks @katrintirok! Looks great. In it goes.

**Comment by ethanwhite** [comment]
The new files have been uploaded to figshare and should be available at:
https://figshare.com/articles/Portal_Project_Teaching_Database/1314459

When a file gets replaced on figshare it gets a new link, so links in existing material will need to be updated to reflect this change.

Thanks again for catching this and contributing the fixes!

**Comment by katrintirok** [comment]
Great, I will check all links in the data carpentry lessons for python and R and submit changes accordingly

**Pull request merged**